migrations/234_submissions__regradable__add.sql

```diff
@@ -1,4 +1,4 @@
-ALTER TABLE submissions ADD COLUMN IF NOT EXISTS regradable boolean NOT NULL DEFAULT FALSE;
+ALTER TABLE submissions ADD COLUMN IF NOT EXISTS regradable boolean DEFAULT FALSE;
 
 UPDATE submissions
 SET regradable = (graded_at IS NOT NULL) OR (grading_requested_at IS NOT NULL);
```

Comment:
@mwest1066 if the concern is updating the whole table, isn't it better to remove this update too?

Reply:
The real problem is not the full table rewrite, but the fact that the table is locked as `ACCESS EXCLUSIVE` while `ALTER TABLE` is running, which blocks all other accesses (including reads and writes). This basically freezes PL while the table is being rewritten. In contrast, `UPDATE` only needs the `ROW EXCLUSIVE` [table lock](https://www.postgresql.org/docs/current/explicit-locking.html), which permits reads and writes to continue. At least this is my understanding 😄